### PR TITLE
Add FXIOS-4937 [v106] Homepage header logo improvements

### PR DESCRIPTION
**Pull request by electricRGB**
[JIRA Reference](https://mozilla-hub.atlassian.net/browse/FXIOS-4937)
#11916

### CODE REVIEW

File: Client/Frontend/Home/LogoHeader/HomeLogoHeaderCell.swift [hunk 1]
```diff
@@ -55,29 +51,38 @@ class HomeLogoHeaderCell: UICollectionViewCell, ReusableCell {
...
 
         NSLayoutConstraint.activate([
-            logoButton.topAnchor.constraint(equalTo: contentView.topAnchor, constant: 32),
-            logoButton.widthAnchor.constraint(equalToConstant: UX.imageWidth),
-            logoButton.heightAnchor.constraint(equalToConstant: UX.imageHeight),
-            logoButton.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
-            logoButton.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -10)
-        ])
-    }
+            logoImage.topAnchor.constraint(equalTo: contentView.topAnchor, constant: 32),
```
[the remark this answers is not in the document]
Comment (electricRGB) [reply Client/Frontend/Home/LogoHeader/HomeLogoHeaderCell.swift]: I DID IT!

**Review by yoanarios — APPROVED** [review]

File: Client/Frontend/Home/LogoHeader/HomeLogoHeaderCell.swift [hunk 2]
```diff
@@ -8,27 +8,32 @@ import UIKit
...
-            config.title = ""
-            config.contentInsets = .zero
-            button.configuration = config
-        } else {
-            button.setTitle("", for: .normal)
-            button.backgroundColor = .clear
-            button.contentEdgeInsets = .zero
-        }
+    lazy var logoImage: UIImageView = .build { imageView in
+        imageView.image = UIImage(imageLiteralResourceName: "fxHomeHeaderLogoBall")
```
Comment (thatswinnie) [inline Client/Frontend/Home/LogoHeader/HomeLogoHeaderCell.swift:29]: Could you please add this to the image identifiers?

File: Client/Frontend/Home/LogoHeader/HomeLogoHeaderCell.swift [hunk 3]
```diff
@@ -55,29 +60,42 @@ class HomeLogoHeaderCell: UICollectionViewCell, ReusableCell {
...
 // MARK: - Theme
 extension HomeLogoHeaderCell: NotificationThemeable {
     func applyTheme() {
-        let resourceName = "fxHomeHeaderLogo"
-        let resourceNameDark = "fxHomeHeaderLogo_dark"
-        let imageString = LegacyThemeManager.instance.currentName == .dark ? resourceNameDark : resourceName
-        logoButton.setImage(UIImage(imageLiteralResourceName: imageString), for: .normal)
+        let wallpaperManager = WallpaperManager()
+        if let logoTextColor = wallpaperManager.currentWallpaper.logoTextColor {
+            logoTextImage.image = UIImage(imageLiteralResourceName: "fxHomeHeaderLogoText")
```
Comment (thatswinnie) [inline Client/Frontend/Home/LogoHeader/HomeLogoHeaderCell.swift:91]: Same here (image identifier)

File: Client/Frontend/Home/LogoHeader/HomeLogoHeaderCell.swift [hunk 4]
```diff
@@ -55,29 +60,42 @@ class HomeLogoHeaderCell: UICollectionViewCell, ReusableCell {
...
+                .withRenderingMode(.alwaysTemplate)
+            logoTextImage.tintColor = logoTextColor
+        } else {
+            logoTextImage.image = UIImage(imageLiteralResourceName: "fxHomeHeaderLogoText")
```
Comment (thatswinnie) [inline Client/Frontend/Home/LogoHeader/HomeLogoHeaderCell.swift:95]: And here (image identifier)

**Review by thatswinnie — APPROVED** [review]

**Pull request merged**